LearningMissionSimulation/PlayGroundGavril.cs

```diff
@@ -39,20 +50,100 @@ public  void CreateAccounts(int accountCount)
                         pendingAccountList.Add(account);
                     }
                 }
-             
             }
         }
+
         public void ActivateAccounts()
         {
             foreach (var account  in pendingAccountList )
             {
                 account.Status = Status.Active; 
                 pendingAccountList.Remove(account);
             }
-            
         }
-    }
-}
 
+        public void CreateSubjects(int subjectCount)
+        {
+            Console.WriteLine("******** Started creating subjects  ********\n");
+            int i = 0;
+            while (i < subjectCount)
+            {
+                Subject subject = ObjectGenerator.GenerateSubject();
+                Console.WriteLine("========== Created  Subject {0} =======\n\n{1}\n", i, subject);
+                subjectList.Add(subject);
+                subjectIdList.Add(subject.Id);
+                i++;
+            }
+            Console.WriteLine("========== Generated {0} Subjects  =======\n", subjectCount);
+            Console.WriteLine("******** Finished creating subjects  ********\n");
+        }
 
+        public void CreateModules(int moduleCount)
+        {
+            Console.WriteLine("******** Started creating modules  ********\n");
+            int i = 0;
+            while (i < moduleCount)
+            {
+                Guid subjectId = subjectIdList[AttributeGenerator.random.Next(0, subjectIdList.Count - 1)];
```

Comment:
The upper limit should be subjectIdList.Count. When we generate random value the upper limit is excluded from the generation.